drivers/sensor/vl53l1x/vl53l1_platform.c

```diff
@@ -0,0 +1,244 @@
+/* vl53l1_platform.c - Driver for ST VL53L1X time of flight sensor */
+
+/*
+ * Copyright (c) 2017 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include "vl53l1_platform.h"
+
+#include <sensor.h>
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <i2c.h>
+#include <logging/log.h>
+
+#define LOG_LEVEL CONFIG_SENSOR_LOG_LEVEL
+LOG_MODULE_DECLARE(VL53L1X);
+
+VL53L1_Error VL53L1_WriteMulti(VL53L1_DEV Dev, uint16_t index, uint8_t *pdata,
+			       uint32_t count)
+{
+	VL53L1_Error Status = VL53L1_ERROR_NONE;
+	int32_t status_int = 0;
+	uint8_t buf[count + 2];
+
+	buf[0] = index >> 8;
+	buf[1] = index & 0xFF;
+	memcpy(&buf[2], pdata, count);
+
+	status_int = i2c_write(Dev->i2c, buf, count + 2, Dev->I2cDevAddr);
+
+	if (status_int < 0) {
+		Status = VL53L1_ERROR_CONTROL_INTERFACE;
+		LOG_ERR("Failed to write");
+	}
+
+	return Status;
+}
+
+VL53L1_Error VL53L1_ReadMulti(VL53L1_DEV Dev, uint16_t index, uint8_t *pdata,
+			      uint32_t count)
```

Comment:
Same here.
What is the difference with VL53L0?
index is on 16-bit instead of 8-bit?

Reply:
Yes, the VL53L1 index is on 16-bit instead of 8-bit

I have thought about re-use same driver but looks the internal implication is different between vl53l1x and vl53l0x, as you already have seen the "index"(register address) length is not the same. Also, those API related to interrupt and threshold and so on. So there should be lots of ifdef, if we want to merge those drivers together.

Reply:
> I have thought about re-use same driver but looks the internal implication is different between vl53l1x and vl53l0x, as you already have seen the "index"(register address) length is not the same. Also, those API related to interrupt and threshold and so on. So there should be lots of ifdef, if we want to merge those drivers together.

Yes, yes. In fact my initial idea was to re-use same driver, but the more I was going deep thru the review the less I found it applicable. So at the end I left it as a comment from your side.